[cppcheck] fix & enable performance checks

Rechi opened the pull request:
## Description
fix & enable cppcheck performance checks

## How has this been tested?
compiled with cppcheck enabled for Android, iOS, Linux, macOS and tvOS

## Types of change
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

garbear reviewed (approved):
I approve the game change. And the rest of the commits look in order. Thanks!

OBorce commented on xbmc/windowing/wayland/WinSystemWayland.cpp:
```diff
@@ -774,7 +774,7 @@ void CWinSystemWayland::ProcessMessages()
 }
 
-void CWinSystemWayland::ApplyShellSurfaceState(IShellSurface::StateBitset state)
+void CWinSystemWayland::ApplyShellSurfaceState(const IShellSurface::StateBitset& state)
```
from what I am seeing the StateBitset is a std:::bitset<5> which is just 8 bytes, so it should be actually cheaper to pass by value then by reference?

yol replied:
Yeah these changes do not really seem useful

ksooo commented on xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxCDDA.h:
```diff
@@ -41,7 +41,7 @@ class CDVDDemuxCDDA : public CDVDDemux
 protected:
   friend class CDemuxStreamAudioCDDA;
-  std::shared_ptr<CDVDInputStream> m_pInput;
+  std::shared_ptr<CDVDInputStream> m_pInput = nullptr;
```
No need to init, not a raw pointer.

ksooo commented on xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxClient.h:
```diff
@@ -52,8 +52,8 @@ class CDVDDemuxClient : public CDVDDemux
-  std::shared_ptr<CDVDInputStream> m_pInput;
-  std::shared_ptr<CDVDInputStream::IDemux> m_IDemux;
+  std::shared_ptr<CDVDInputStream> m_pInput = nullptr;
+  std::shared_ptr<CDVDInputStream::IDemux> m_IDemux = nullptr;
```
No need to init, not a raw pointers.

ksooo commented on xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.h:
```diff
@@ -112,7 +112,7 @@ class CDVDDemuxFFmpeg : public CDVDDemux
 
   AVFormatContext* m_pFormatContext;
-  std::shared_ptr<CDVDInputStream> m_pInput;
+  std::shared_ptr<CDVDInputStream> m_pInput = nullptr;
```
No need to init, not a raw pointer.

ksooo commented on xbmc/cores/VideoPlayer/DVDMessageQueue.h:
```diff
@@ -19,22 +19,17 @@
   DVDMessageListItem& operator=(const DVDMessageListItem&) = delete;
 
-  std::shared_ptr<CDVDMsg> message;
+  std::shared_ptr<CDVDMsg> message = nullptr;
```
No need to init, not a raw pointer.

ksooo commented on xbmc/cores/VideoPlayer/VideoPlayerVideo.h:
```diff
@@ -133,7 +133,7 @@ class CVideoPlayerVideo : public CThread, public IDVDStreamPlayerVideo
   CDVDMessageQueue& m_messageParent;
   CDVDStreamInfo m_hints;
-  std::unique_ptr<CDVDVideoCodec> m_pVideoCodec;
+  std::unique_ptr<CDVDVideoCodec> m_pVideoCodec = nullptr;
```
No need to init, not a raw pointer.

ksooo commented on xbmc/cores/paplayer/VideoPlayerCodec.h:
```diff
@@ -42,8 +42,8 @@ class VideoPlayerCodec : public ICodec
   CDVDDemux* m_pDemuxer;
-  std::shared_ptr<CDVDInputStream> m_pInputStream;
-  std::unique_ptr<CDVDAudioCodec> m_pAudioCodec;
+  std::shared_ptr<CDVDInputStream> m_pInputStream = nullptr;
```
No need to init, not raw pointers.

ksooo commented on xbmc/dialogs/GUIDialogKeyboardGeneric.h:
```diff
@@ -80,7 +80,7 @@ class CGUIDialogKeyboardGeneric : public CGUIDialog, public CGUIKeyboard
     std::string m_strHeading;
     std::string m_text;       ///< current text
 
-    IInputCodingTablePtr m_codingtable;
```
Is this a raw pointer?

ksooo commented on xbmc/input/KeyboardLayout.h:
```diff
@@ -48,5 +48,5 @@ class CKeyboardLayout
   std::string m_layout;
   Keyboards m_keyboards;
-  IInputCodingTablePtr m_codingtable;
+  IInputCodingTablePtr m_codingtable = nullptr;
```
Is this a raw pointer?

Rechi commented:
@ksooo I've added a fixup commit which removes the default initialization with `nullptr`. I'll squash it, if evertything is okay.
@yol I've dropped the `IShellSurface::StateBitset` changes.

yol reviewed (approved):
wayland changes are gone, but I didn't have the time to look into the rest

thexai commented:
jenkins build this please

ksooo reviewed (approved):
LGTM, as soon as clang-format is happy.

Rechi committed:
[cppcheck] postfixOperator

Rechi committed:
[cppcheck] useInitializationList

Rechi committed:
[cppcheck] stlcstrParam

Rechi committed:
[cppcheck] passedByValue

Rechi committed:
[cppcheck] stlFindInsert

Rechi committed:
[cppcheck] enable performance checks
